### ISSUE TITLE
rustc fails with unknown feature `proc_macro_span_shrink`

### ISSUE THREAD

**Issue by dubek** [rust-lang/rust#113170]
I tried to build this simple project (mainly I tried to compile the `proc-macro2` crate):

Cargo.toml:

```toml
[package]
name = "t1"
version = "0.1.0"
edition = "2021"

[dependencies]
proc-macro2 = "=1.0.56"
```

rust-toolchain.toml:
```
[toolchain]
channel = "nightly"
```

src/main.rs:

```rust
fn main() {
    println!("Hello, world!");
}
```

I expected to see this happen: it compiles

Instead, this happened: 

```
$ cargo build
   Compiling proc-macro2 v1.0.56
   Compiling unicode-ident v1.0.9
error[E0635]: unknown feature `proc_macro_span_shrink`
  --> /home/dov/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.56/src/lib.rs:92:30
   |
92 |     feature(proc_macro_span, proc_macro_span_shrink)
   |                              ^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0635`.
error: could not compile `proc-macro2` (lib) due to previous error
```

### Meta

`rustc --version --verbose`:
```
rustc 1.72.0-nightly (5bd28f5ea 2023-06-28)
binary: rustc
commit-hash: 5bd28f5eac1ba3569bfa8d49ec3f5acbdfdff7a0
commit-date: 2023-06-28
host: x86_64-unknown-linux-gnu
release: 1.72.0-nightly
LLVM version: 16.0.5
```

<details><summary>Backtrace</summary>
<p>

```
   Compiling proc-macro2 v1.0.56
   Compiling unicode-ident v1.0.9
error[E0635]: unknown feature `proc_macro_span_shrink`
  --> /home/dov/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.56/src/lib.rs:92:30
   |
92 |     feature(proc_macro_span, proc_macro_span_shrink)
   |                              ^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0635`.
error: could not compile `proc-macro2` (lib) due to previous error
```

</p>
</details>

**Comment by dubek** [comment]
hmm, it seems that this is solved in proc_macro2 1.0.60 (and newer).

so maybe not a rustc problem?

**Comment by lqd** [comment]
Yes, this is https://github.com/rust-lang/rust/issues/113152
